universal_silabs_flasher/flasher.py

```diff
@@ -221,6 +239,8 @@ async def probe_app_type(
                 # We cannot assume that the bootloader is the only running application
                 # if we did not try to start an application
                 if only_probe_bootloader:
+                    if self.app_type is None:
```

Comment:
I think the entire branch (`if only_probe_bootloader`) can actually be removed, since `result.continue_probing` will be `False` when we only probe the bootloader.

Reply:
I was trying to avoid not overwriting the app_type here, in the case when this function gets called a second time to only probe the bootloader baudrate, however I think this happens late enough that it doesn't actually matter.  So I have removed this `if only_probe_bootloader` block.

I have also tweaked the logic in the "invalid application" case a few lines further down to be a bit clearer. The `only_probe_bootloader` condition here is not actually needed and may introduce a corner case failure (when manually enter bootloader on device with invalid application and no reset flag provided)